accepted/wasmtime-rpc.md

```diff
@@ -0,0 +1,162 @@
+
+- [`bytes`](https://docs.rs/bytes/latest/bytes/)
+- [`futures`](https://docs.rs/futures/latest/futures/)
+- [`tokio-stream`](https://docs.rs/tokio-stream/latest/tokio_stream/)
+- [`tokio-util`](https://docs.rs/tokio-util/latest/tokio_util/)
+- [`tokio`](https://docs.rs/tokio/latest/tokio/)
+
+The only external wRPC dependency not already used by Wasmtime is [`pin-project-lite`](https://docs.rs/pin-project-lite/latest/pin_project_lite/).
+Other than that, wRPC depends on a few small, purpose-built crates maintained by wRPC developers and extracted from wRPC codebase itself.
+
+## Bytecode Alliance hosted project
+
+[wRPC] is a Bytecode Alliance hosted project and so Bytecode Alliance has full control over the project's development.
+
+For example, if the `pin-project-lite` dependency mentioned above is an issue - that can be promptly changed.
+
+## WIT as lingua franca
+
+[wRPC] is built on top of WIT, which is already utilizied throughout Wasmtime codebase and, in fact, is embedded in each and every component, therefore making each component compatible with [wRPC] out of the box.
+There is no need for a custom mapping scheme and/or naming convention as one would need with a different RPC protocol (e.g. gRPC).
+
+## WebAssembly as the wire format
+
+[wRPC] relies on [component model value definition encoding], which in turn is based on Core Wasm spec and Canonical ABI.
+In fact, it can be thought of as a "packed" Canonical ABI, optimized for data size.
+
+Usage of a familiar encoding here provides many optimization opportunities on the runtime level providing for efficient serialization and deserialization steps.
+
+## Familiar user experience
+
+[wRPC] provides `wit-bindgen-wrpc` binary and Rust crate, which are built on top of `wit-bindgen`, which provides a familiar user experience.
+For example, `wit_bindgen_wrpc::generate!` macro supports most of the options supported by `wit_bindgen::generate!`.
+
+Plugin developers familiar with component model would not need to learn a new technology to develop a Wasmtime plugin, rather use largely the same development flow that is already used for developing components.
+
+## Multi-language support
+
+Other than Rust, [wRPC] currently supports Go, but support for other languages is planned.
+Language binding generators benefit greatly from prior art in `wit-bindgen`, since a lot of that logic is reused.
+
+## Transport-agnostic
+
+[wRPC] is transport-agnostic, with 4 supported transports at the time of writing:
+- TCP
+- Unix domain sockets
+- QUIC
+- [NATS.io](https://nats.io/)
+
+Only TCP and Unix domain socket transports are suggested to use for Wasmtime CLI in this RFC to avoid pulling in extra dependencies.
+
+# Open questions
+[open-questions]: #open-questions
+
+- How do plugin imports work together with host builtins? I think the right solution here is to throw an error if e.g. both built-in `wasi:keyvalue` *and* `wasi:keyvalue` plugin import are specified.
```

Comment:
The only option that would allow backward compatibility on the Wasmtime side I think would be to make plugins shadow builtin implementations.